UPGRADE.rst

```diff
@@ -55,50 +55,140 @@ Upgrading to v1.4.0
 
-If you have configured a custom template directory with the ``email.template_dir`` option, be
-aware that there are new templates regarding registration. ``registration.html`` and
-``registration.txt`` have been added and contain the content that is sent to a client upon
-registering via an email address.
+If you have configured a custom template directory with the
+``email.template_dir`` option, be aware that there are new templates regarding
+registration and threepid management.
+
+``registration.html`` and ``registration.txt`` contain the content that is sent
+to a client upon registering via an email address.
+
+``registration_success.html`` and ``registration_failure.html`` are templates
+that will be shown to the user when they click the link in their registration
+email , either showing them a success or failure page (assuming a redirect URL
+is not configured).
 
-``registration_success.html`` and ``registration_failure.html`` are also new HTML templates
-that will be shown to the user when they click the link in their registration emai , either
-showing them a success or failure page (assuming a redirect URL is not configured).
+``add_threepid.html`` and  ``add_threepid.txt`` are templates sent to a client
+to add the threepid.
 
-Synapse will expect these files to exist inside the configured template directory. To view the
-default templates, see `synapse/res/templates
+``add_threepid_failure.html`` and ``add_threepid_success.html`` are templates
+that will be shown to the user when they click the link in their threeid add
+email , either showing them a success or failure page (assuming a redirect URL
+is not configured).
+
+
+Synapse will expect these files to exist inside the configured template
+directory. To view the default templates, see `synapse/res/templates
 <https://github.com/matrix-org/synapse/tree/master/synapse/res/templates>`_.
 
+SMS
+---
+
+Following upgrade, the only way to maintain the ability to register via a phone
+number will be to continue to delegate SMS delivery via the matrix.org and
+vector.im identity servers.
+
+The ``account_threepid_delegates`` dictionary defines whether the homeserver
+should delegate an external server (typically an `identity server
+<https://matrix.org/docs/spec/identity_service/r0.2.1>`_) to handle sending
+password reset or registration messages via email and SMS.
+
+So to delegate SMS sending set ``account_threepid_delegates.sms`` to a base URL of
+an identity server in your homeserver.yaml.
+
+.. code:: yaml
+
+   account_threepid_delegates:
+       msisdn: https://example.com     # Delegate sms sending to example.com
+
+Currently Synapse does not support a means to send SMS itself, and the
+matrix.org and vector.im identity servers will continue to support SMS until
+such time as it is possible for admins to configure their servers to send SMS
+directly. More details will follow in a future release.
+
+
+For more details on why these changes are necessary see (link to blog).
```

Comment:
if this is going to be relevant to both msisdn and email, it should be elsewhere?

Reply:
The idea is that it is at the same level of indentation as 'Upgrading to v1.4.0'. I've moved it higher in the doc, should be clearer now.